buildSrc/src/main/kotlin/kord-module.gradle.kts

```diff
@@ -34,16 +37,14 @@ atomicfu {
     dependenciesVersion = libs.findVersion("kotlinx-atomicfu").get().requiredVersion
 }
 
-tasks {
-    withType<JavaCompile> {
-        sourceCompatibility = Jvm.targetString
-        targetCompatibility = Jvm.targetString
-    }
+// the kspKotlin Task is created very late and isn't available via tasks.findByName("kspKotlin")
+val kspKotlin get() = tasks.withType<KspTask>().matching { it.name == "kspKotlin" }
```

Comment:
Do we know why this happens? It's very weird, that your method filtering by name works and findByName doesn't?

Reply:
I searched in the sources of the ksp gradle plugin and it seems to not register its tasks when the plugin is [applied](https://github.com/google/ksp/blob/1a1818bf6524665c163335ab6e6d70d8ded6167f/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt#L229) but instead [here](https://github.com/google/ksp/blob/1a1818bf6524665c163335ab6e6d70d8ded6167f/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt#L383) inside the [`applyToCompilation`](https://github.com/google/ksp/blob/1a1818bf6524665c163335ab6e6d70d8ded6167f/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt#L260) function overriding the one in `KotlinCompilerPluginSupportPlugin`, whenever that is called.
The reason why `findByName` doesn't work is that it is not one of the [Task Configuration Avoidance](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html) APIs that returns a `TaskProvider` but instead returns `Task?` directly. `tasks.withType<KspTask>().matching { it.name == "..." }` seems to be the only lazy way to query a task that might not exist (the `kspKotlin` tasks only exists in the subprojects that actually have a ksp processor in their dependencies).

Reply:
Ok I was able to solve this, see https://github.com/kordlib/kord/pull/755/commits/00cbc9e73e3451cd3e49eec4e99a1ae1483e2c16.